service/controller/chart/v1/resource/tiller/resource.go

```diff
@@ -48,7 +48,13 @@ func (r *Resource) Name() string {
 func (r *Resource) ensureTillerInstalled(ctx context.Context) error {
 	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring tiller is installed")
 
-	err := r.helmClient.EnsureTillerInstalled(ctx)
+	values := []string{
+		"spec.template.spec.priorityClassName=giantswarm-critical",
+		"spec.template.spec.tolerations[0].effect=NoSchedule",
+		"spec.template.spec.tolerations[0].key=node-role.kubernetes.io/master",
+		"spec.template.spec.tolerations[0].operator=Exists",
```

Comment:
Can we have comments why `[0]`?

Reply:
I added a very simple comment. Tolerations are just handled like a list and there `[0]` is the first entry into the list of tolerations.

Reply:
That is pretty odd. Can we overwrite something? What if the list is empty? In my head this can fail in so many different ways. 🙈

Reply:
This is a "normal" yaml value overwrite for helm as far as I understand it. The list we pass here is internally cast to a yaml before it is applied from my understanding. 
Leaving the list empty should be the same as having an empty list in yaml e.g.:
```
spec:
  template:
    spec:
      tolerations:
```
So nothing should fail - from my tests this was pretty robust (aka hard to break), but unintuitive to use.

Reply:
Yes this is the "interface" Helm provides for extending the Tiller deployment.

We want to move away from Tiller because it causes us operational pain and the security model is not good. Server side apply is an alpha feature in K8s 1.14. It's one possible route for us using Helm only for templating.